src/test/ui/async-await/async-fn-send-uses-nonsend.rs

```diff
@@ -0,0 +1,59 @@
+// compile-pass
+// edition:2018
+// compile-flags: --crate-type lib
+
+#![feature(async_await)]
+
+use std::{
+    cell::RefCell,
+    fmt::Debug,
+    rc::Rc,
+};
+
+fn non_sync() -> impl Debug { RefCell::new(()) }
+
+fn non_send() -> impl Debug { Rc::new(()) }
+
+fn take_ref<T>(_: &T) {}
+
+async fn fut() {}
+
+async fn fut_arg<T>(_: T) {}
+
+async fn still_send() {
+    fut().await;
+    println!("{:?} {:?}", non_send(), non_sync());
+    fut().await;
+    drop(non_send());
+    drop(non_sync());
+    fut().await;
+    fut_arg(non_sync()).await;
+
+    // Note: all temporaries in `if let` and `match` discriminee
```

Comment:
```suggestion
    // Note: all temporaries in `if let` and `match` scrutinee
```
(https://doc.rust-lang.org/nightly/reference/glossary.html#scrutinee -- the discriminant is the tag of the enum)

Reply:
ah thank you I was trying to remember that word